### PR TITLE
kokkos: add gfx940 support

### PR DESCRIPTION
**Pull request by rbberger**
Should be merged after https://github.com/kokkos/kokkos/pull/6671 gets into kokkos.

### CODE REVIEW

File: var/spack/repos/builtin/packages/kokkos/package.py [hunk 1]
```diff
@@ -238,6 +238,11 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     # Patches
     patch("hpx_profiling_fences.patch", when="@3.5.00 +hpx")
     patch("sycl_bhalft_test.patch", when="@4.2.00 +sycl")
+    patch(
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/kokkos/package.py:244]: Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed

?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request, otherwise LGTM

File: var/spack/repos/builtin/packages/kokkos/package.py (hunk 1 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/kokkos/package.py:244]: Will do. I'm afk until early January due to travel. I'll push it as soon as I can.

Comment (rbberger) [reply var/spack/repos/builtin/packages/kokkos/package.py:244]: @alalazo added the comment.

**Review by alalazo — APPROVED** [review]

**Pull request merged**